File: java/Dockerfile.java
ARG DUMPER_VERSION=1.13 \
    YSOSERIAL_VERSION=0.0.6

FROM eclipse-temurin:17-jdk

LABEL maintainer="support@go-forward.net" \
    vendor="Go Forward"

ARG DUMPER_VERSION \
    YSOSERIAL_VERSION

RUN curl -L -o /usr/local/bin/SerializationDumper.jar "https://github.com/NickstaDB/SerializationDumper/releases/download/${DUMPER_VERSION}/SerializationDumper-v${DUMPER_VERSION}.jar" && \
    curl -L -o /usr/local/bin/ysoserial.jar "https://github.com/frohoff/ysoserial/releases/download/v${YSOSERIAL_VERSION}/ysoserial-all.jar"

WORKDIR /srv/demo/net/gofwd/
COPY Serialize.java .
COPY Deserialize.java .
COPY MyObject.java /srv/demo/net/gofwd/
COPY serialize.sh /usr/local/bin/
COPY deserialize.sh /usr/local/bin/
COPY SerializationDumper.sh /usr/local/bin/
COPY ysoserial.sh /usr/local/bin/

RUN javac MyObject.java Serialize.java Deserialize.java

RUN chmod ugo+x /usr/local/bin/serialize.sh && \
    chmod ugo+x /usr/local/bin/deserialize.sh && \
    chmod ugo+x /usr/local/bin/SerializationDumper.sh && \
    chmod ugo+x /usr/local/bin/ysoserial.sh && \
    ln -s /usr/local/bin/serialize.sh /usr/local/bin/serialize && \
    ln -s /usr/local/bin/deserialize.sh /usr/local/bin/deserialize && \
    ln -s /usr/local/bin/SerializationDumper.sh /usr/local/bin/SerializationDumper && \
    ln -s /usr/local/bin/ysoserial.sh /usr/local/bin/ysoserial

WORKDIR /workdir

File: php/Dockerfile.php
# Use a base image to build (and download) the tools on

FROM php:cli-alpine

LABEL maintainer="support@go-forward.net"
LABEL vendor="Go Forward"

WORKDIR /usr/local/bin/

COPY serialize.php .
COPY serialize_function.php /usr/local/lib/php/
COPY deserialize.php .
COPY deserializeobject.php /usr/local/lib/php/
COPY serialize_chain.php .
COPY serialize_chain_self.php .
COPY deserialize_chain.php .
RUN chmod ugo+x /usr/local/bin/serialize.php && \
    chmod ugo+x /usr/local/bin/serialize_chain.php && \
    chmod ugo+x /usr/local/bin/serialize_chain_self.php && \
    chmod ugo+x /usr/local/bin/deserialize.php && \
    chmod ugo+x /usr/local/bin/deserialize_chain.php && \
    ln -s /usr/local/bin/serialize.php /usr/local/bin/serialize && \
    ln -s /usr/local/bin/serialize_chain.php /usr/local/bin/serialize_chain && \
    ln -s /usr/local/bin/serialize_chain_self.php /usr/local/bin/serialize_chain_self && \
    ln -s /usr/local/bin/deserialize.php /usr/local/bin/deserialize && \
    ln -s /usr/local/bin/deserialize_chain.php /usr/local/bin/deserialize_chain

WORKDIR /workdir
